compilers/concrete-compiler/compiler/lib/Bindings/Python/CompilerAPIModule.cpp

```diff
@@ -923,6 +923,14 @@ void mlir::concretelang::python::populateCompilerAPISubmodule(
            [](CompilationOptions &options, int security_level) {
              options.optimizerConfig.security = security_level;
            })
+      .def("set_parameter_restrictions",
```

Comment:
Just so that you know, `std::optional` are properly recognized by pybind. if you use them in the signature, it will work as expected.